diff -rupN org-libyuv-main/CMakeLists.txt cur-libyuv-main/CMakeLists.txt
--- org-libyuv-main/CMakeLists.txt	2026-08-14 18:06:42.000000000 +0200
+++ cur-libyuv-main/CMakeLists.txt	2026-08-16 16:40:00.000000000 +0200
@@ -171,15 +171,17 @@ endif()
 
 # this creates the static library (.a)
 add_library( ${ly_lib_static} STATIC ${ly_lib_parts})
 
-# this creates the shared library (.so)
-add_library( ${ly_lib_shared} SHARED ${ly_lib_parts})
-set_target_properties( ${ly_lib_shared} PROPERTIES OUTPUT_NAME "${ly_lib_name}" )
-set_target_properties( ${ly_lib_shared} PROPERTIES PREFIX "lib" )
-if(WIN32)
-  set_target_properties( ${ly_lib_shared} PROPERTIES IMPORT_PREFIX "lib" )
-endif()
-
+# this creates the shared library (.so) - only when BUILD_SHARED_LIBS is ON
+if(BUILD_SHARED_LIBS)
+  add_library( ${ly_lib_shared} SHARED ${ly_lib_parts})
+  set_target_properties( ${ly_lib_shared} PROPERTIES OUTPUT_NAME "${ly_lib_name}" )
+  set_target_properties( ${ly_lib_shared} PROPERTIES PREFIX "lib" )
+  if(WIN32)
+    set_target_properties( ${ly_lib_shared} PROPERTIES IMPORT_PREFIX "lib" )
+  endif()
+endif()
+
 # this creates the cpuid tool
 #add_executable      ( cpuid ${ly_base_dir}/util/cpuid.c )
 #target_link_libraries  ( cpuid ${ly_lib_static} )
@@ -244,7 +246,9 @@ endif()
 
 # install the conversion tool, .so, .a, and all the header files
 #install ( TARGETS yuvconvert DESTINATION bin )
 install ( TARGETS ${ly_lib_static}						DESTINATION lib )
-install ( TARGETS ${ly_lib_shared} LIBRARY DESTINATION lib RUNTIME DESTINATION bin ARCHIVE DESTINATION lib )
+if(BUILD_SHARED_LIBS)
+  install ( TARGETS ${ly_lib_shared} LIBRARY DESTINATION lib RUNTIME DESTINATION bin ARCHIVE DESTINATION lib )
+endif()
 install ( DIRECTORY ${PROJECT_SOURCE_DIR}/include/		DESTINATION include )
