diff -rupN smalltalk-3.2.5/libgst/sym.h patched/libgst/sym.h
--- smalltalk-3.2.5/libgst/sym.h	2013-04-07 19:24:12.022020096 +0000
+++ patched/libgst/sym.h	2017-10-18 12:46:45.061865984 +0000
@@ -372,7 +372,7 @@ extern struct builtin_selector _gst_buil
   ATTRIBUTE_HIDDEN;
 
 extern struct builtin_selector *_gst_lookup_builtin_selector (const char *str,
-							      unsigned int len)
+							      size_t len)
   ATTRIBUTE_HIDDEN;
 
 #endif /* GST_SYM_H */
