diff -Naur openal-soft-1.17.2_org/CMakeLists.txt openal-soft-1.17.2/CMakeLists.txt
--- openal-soft-1.17.2_org/CMakeLists.txt	2017-07-10 10:15:15.798752768 +0000
+++ openal-soft-1.17.2/CMakeLists.txt	2017-07-10 10:16:20.143654912 +0000
@@ -162,10 +162,11 @@
 
     CHECK_C_SOURCE_COMPILES("extern inline int foo() { return 0; }
                              int main() {return foo();}" INLINE_IS_C99)
-    IF(NOT INLINE_IS_C99)
-        MESSAGE(FATAL_ERROR "Your compiler does not seem to have C99 inline semantics!
-                             Please update your compiler for better C99 compliance.")
-    ENDIF()
+    # Haiku has GCC 5.4.0 but this check fails
+    #IF(NOT INLINE_IS_C99)
+    #    MESSAGE(FATAL_ERROR "Your compiler does not seem to have C99 inline semantics!
+    #                         Please update your compiler for better C99 compliance.")
+    #ENDIF()
 
     SET(CMAKE_REQUIRED_FLAGS "${OLD_REQUIRED_FLAGS}")
 ENDIF()
