diff -rupN openal-soft-1.17.2_org/OpenAL32/Include/alMain.h openal-soft-1.17.2/OpenAL32/Include/alMain.h
--- openal-soft-1.17.2_org/OpenAL32/Include/alMain.h	2017-07-10 14:31:00.153616384 +0000
+++ openal-soft-1.17.2/OpenAL32/Include/alMain.h	2017-07-10 14:39:28.037224448 +0000
@@ -54,15 +54,7 @@ ALC_API void ALC_APIENTRY alcGetInteger6
 typedef ALint64SOFT ALint64;
 typedef ALuint64SOFT ALuint64;
 
-#ifndef U64
-#if defined(_MSC_VER)
-#define U64(x) ((ALuint64)(x##ui64))
-#elif SIZEOF_LONG == 8
-#define U64(x) ((ALuint64)(x##ul))
-#elif SIZEOF_LONG_LONG == 8
 #define U64(x) ((ALuint64)(x##ull))
-#endif
-#endif
 
 #ifndef UINT64_MAX
 #define UINT64_MAX U64(18446744073709551615)
