diff --git a/SConstruct b/SConstruct
index 0a24811..0916ac9 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1,5 +1,7 @@
 from ternary_variable import TernaryVariable
 from glob_files import globFiles
+import os
+import sys
 
 # create parent environment and load tools
 globalEnv = Environment()
@@ -20,9 +22,6 @@ globalEnv.configReport.add('Target platform', 'target_platform')
 
 targetPlatform = globalEnv['target_platform']
 if targetPlatform == 'auto':
-	import os
-	import sys
-	
 	if sys.platform == 'haiku1':
 		targetPlatform = 'beos'
 	elif os.name == 'win32' or sys.platform == 'win32':
@@ -40,6 +39,12 @@ if targetPlatform == 'windows':
 	globalEnv['WINDOWS_INSERT_DEF'] = 1 # generated .def file if dll is build
 	globalEnv.Append(SHLINKFLAGS = ['-Wl,--export-all-symbols']) # windows requires this
 
+# Haiku: The PATH found by SCons are wrong in many ways causing binaries to be not found.
+#        Replace them with sane values. This is not a 'good' solution but should work.
+if sys.platform == 'haiku1':
+	import os
+	globalEnv['ENV']['PATH'] = os.environ['PATH']
+
 # parameters
 params = Variables(globalEnv['PARAMETER_SOURCE'], ARGUMENTS)
 params.Add(BoolVariable('with_debug', 'Build with debug symbols for GDB usage', False))
