diff -rupN org_smalltalk-3.2.5/lib-src/getdtablesize.c new_smalltalk-3.2.5/lib-src/getdtablesize.c
--- org_smalltalk-3.2.5/lib-src/getdtablesize.c	2013-03-23 20:56:26.000000000 +0100
+++ new_smalltalk-3.2.5/lib-src/getdtablesize.c	2017-05-15 15:16:56.769397341 +0200
@@ -50,6 +50,15 @@
  *
  ***********************************************************************/
 
+#ifdef DE_ANDROID
+
+#include <sys/sysconf.h>
+int getdtablesize(void) {
+	return sysconf(_SC_OPEN_MAX);
+}
+
+#else
+
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
@@ -107,3 +116,4 @@ getdtablesize(void)
 }
 #endif
 
+#endif
diff -rupN org_smalltalk-3.2.5/lib-src/getpagesize.c new_smalltalk-3.2.5/lib-src/getpagesize.c
--- org_smalltalk-3.2.5/lib-src/getpagesize.c	2013-03-23 20:56:26.000000000 +0100
+++ new_smalltalk-3.2.5/lib-src/getpagesize.c	2017-05-15 15:10:33.394355445 +0200
@@ -50,6 +50,8 @@
  *
  ***********************************************************************/
 
+#ifndef DE_ANDROID
+
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
@@ -106,3 +108,5 @@ getpagesize(void)
 #endif /* no _SC_PAGE_SIZE */
 }
 #endif
+
+#endif
