diff -rupN org-openal-soft-1.22.2/core/filters/biquad.cpp openal-soft-1.22.2/core/filters/biquad.cpp
--- org-openal-soft-1.22.2/core/filters/biquad.cpp	2022-06-27 10:33:53.000000000 +0200
+++ openal-soft-1.22.2/core/filters/biquad.cpp	2022-12-09 21:23:40.481301066 +0100
@@ -15,7 +15,7 @@ template<typename Real>
 void BiquadFilterR<Real>::setParams(BiquadType type, Real f0norm, Real gain, Real rcpQ)
 {
     // Limit gain to -100dB
-    assert(gain > 0.00001f);
+    gain = gain > 0.00001f ? gain : 0.00001f;
 
     const Real w0{al::numbers::pi_v<Real>*2.0f * f0norm};
     const Real sin_w0{std::sin(w0)};
diff -rupN org-openal-soft-1.22.2/core/voice.h openal-soft-1.22.2/core/voice.h
--- org-openal-soft-1.22.2/core/voice.h	2022-06-27 10:33:53.000000000 +0200
+++ openal-soft-1.22.2/core/voice.h	2022-12-09 21:24:03.978430005 +0100
@@ -32,7 +32,7 @@ enum class DistanceModel : unsigned char
 using uint = unsigned int;
 
 
-#define MAX_SENDS  6
+#define MAX_SENDS  64
 
 
 enum class SpatializeMode : unsigned char {
